src/features/courses/course-controller.ts

```diff
@@ -2454,9 +2456,10 @@ class CourseController {
                     await questions.asyncForEach(async (question) => {
                         if (_.isNil(question.grades)) throw new RederlyExtendedError('Impossible! Found an assessment question without a grade.');
                         const version = await courseRepository.getCurrentInstanceForQuestion({questionId: question.id, userId});
-                        question.webworkQuestionPath = version?.webworkQuestionPath ?? question.webworkQuestionPath;
-                        question.grades.randomSeed = version?.randomSeed ?? question.grades.randomSeed;
-                        question.problemNumber = version?.problemNumber ?? question.problemNumber;
+                        if (!_.isNil(version)){
+                            question.grades[0].getVersion(version);
+                            question.getVersion(version);
```

Comment:
I don't really understand what's happening here.

Reply:
this is what Tom asked for -- using the methods on CourseWWTopicQuestion to selectively replace fields according to a StudentGradeInstance...

Reply:
I'm reverting it anyhow